packages/typespec-ts/test/commands/run.js

```diff
@@ -1,67 +1,181 @@
+import { fileURLToPath } from "url";
 import { dirname, join as joinPath } from "path";
+import { Extractor, ExtractorConfig, ExtractorLogLevel } from "@microsoft/api-extractor";
 import { npxCommand } from "./runCommand.js";
-import { fileURLToPath } from "url";
-import { access } from "fs/promises";
+import * as fs from "fs/promises";
 import { createTaskLogger } from "./logger.js";
-
+import { createProgram } from "typescript";
+export async function runTypespec(config, mode) {
+    await run(createContext());
+    function createContext() {
+        const { inputPath: sourceTypespec, outputPath: targetFolder } = config;
+        const __filename = fileURLToPath(import.meta.url);
+        const testRoot = dirname(__filename);
+        const logger = createTaskLogger();
+        return {
+            logger: () => logger,
+            mode: () => mode,
+            sourceTypespec: () => sourceTypespec,
+            targetFolder: () => targetFolder,
+            testRoot: () => testRoot
+        };
+    }
+}
+async function run(ctx) {
+    await (async () => {
+        const logger = ctx.logger();
+        logger.log(`=== Start ${ctx.targetFolder()} ===`);
+        try {
+            await emitClient();
+            logger.log("=== Emitting declaration files ===");
+            await emitDeclarationFiles();
+            logger.log("=== Emitting API summary ===");
+            await emitAPISummary();
+            logger.log(`=== End ${ctx.targetFolder()} ===`);
+        }
+        catch (e) {
+            logger.error(e.toString());
+            logger.flush();
+            throw e;
+        }
+        finally {
+            logger.flush();
+        }
+    })();
+    async function emitClient() {
+        const logger = ctx.logger();
+        const workingDir = outputPath();
+        const commandArguments = [
+            "compile",
+            `${await entryPath()}`,
+            "--config tspconfig.yaml "
+        ];
+        logger.flush();
+        await npxCommand("tsp", commandArguments, workingDir, logger);
+    }
+    async function emitDeclarationFiles() {
+        const logger = ctx.logger();
+        const program = createProgram({
+            options: {
+                declaration: true,
+                emitDeclarationOnly: true,
+                declarationMap: true,
+                removeComments: true,
+                rootDir: outputPath()
+            },
+            rootNames: [joinPath(outputPath(), "src/index.ts")]
+        });
+        // side effect loads source files into memory
+        // nothing will be emitted if this is omitted
+        program.getSourceFiles();
```

Comment:
I wonder how we run the integration test with ts-morph program to load all the generated files? 
Also, want to confirm if we are not running the test parallelly ?